terminal/src/Format.hs

```diff
@@ -133,6 +135,52 @@ format flags (Env inputs) =
     Project projectType paths ->
       formatFilesOnDisk flags projectType paths
 
+validate :: Env -> Task.Task Exit.Format ()
+validate (Env inputs) = do
+  case inputs of
+    Stdin ->
+      do
+        original <- Task.io BS.getContents
+        case formatByteString Parse.Application original of
+          Nothing ->
+            error "TODO: report error"
+          Just formatted ->
+            when (BSL.fromStrict original /= B.toLazyByteString formatted) $
+              Task.throw Exit.FormatValidateNotCorrectlyFormatted
+    Files paths ->
+      validateFiles paths
+    Project _ paths ->
+      validateFiles paths
+
+validateFiles :: [FilePath] -> Task.Task Exit.Format ()
+validateFiles paths = do
+  validationResults <- mapM validateFile paths
+  when (any (== False) validationResults) $
+    Task.throw Exit.FormatValidateNotCorrectlyFormatted
```

Comment:
Currently I check all files before throwing an error so you get a full report.
Is it better to just throw on first invalid file?

Reply:
Personally, I think checking all files is good. I often use `gren format --validate` in CI settings, so I don't necessarily care about how long the validation takes.

Reply:
Yeah, checking all seems best, given that this ought to be fast even with lots of files.